core/src/com/unciv/logic/GameStarter.kt

```diff
@@ -222,7 +222,22 @@ object GameStarter {
     private fun addCivilizations(newGameParameters: GameParameters, gameInfo: GameInfo, ruleset: Ruleset, existingMap: Boolean) {
         val availableCivNames = Stack<String>()
         // CityState or Spectator civs are not available for Random pick
-        availableCivNames.addAll(ruleset.nations.filter { it.value.isMajorCiv() }.keys.shuffled())
+        if (gameSetupInfo.mapParameters.blacklistRandomNationsPool) {
```

Comment:
Surely the 'enable' if should come before the 'blacklist' if... that's the higher-level function enabler